boskos/janitor/janitor.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"os/exec"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"k8s.io/test-infra/boskos/client"
+)
+
+var (
+	clean    = janitorClean
+	POOLSIZE = 10 // Maximum concurrent janitor goroutines
+)
+
+func main() {
+	logrus.SetFormatter(&logrus.JSONFormatter{})
+	boskos := client.NewClient("Janitor", "http://boskos")
+	logrus.Info("Initialized boskos client!")
+	janitorPool := make(chan string, 1)
+
+	for i := 0; i < POOLSIZE; i++ {
+		go janitor(boskos, janitorPool)
+	}
+
+	for {
+		if proj, err := boskos.Acquire("project", "dirty", "cleaning"); err != nil {
+			logrus.WithError(err).Error("Boskos acquire failed!")
+			time.Sleep(time.Minute)
+		} else if proj == "" {
+			time.Sleep(time.Minute)
+		} else {
+			janitorPool <- proj // will block until pool has a free slot
+		}
+	}
+}
+
+// Clean by janitor script
+func janitorClean(proj string) error {
+	script := "../../jenkins/janitor.py"
+	return exec.Command(fmt.Sprintf("%s --project=%s --hour=0", script, proj)).Run()
```

Comment:
Have you run this? I'm pretty sure this needs to be `exec.Command("janitor.py", "--project=P", "--hour=0")` There is no file named `janitor.py --project=P --hour=0`

https://golang.org/pkg/os/exec/#Command